pkg/virtctl/imageupload/imageupload.go

```diff
@@ -386,6 +390,27 @@ func (c *command) run(args []string) error {
 		return err
 	}
 
+	if dataSource {
```

Comment:
The control flow is a bit unclear here. What about this:

```suggestion
	if dataSource {
		ds, err := virtClient.CdiClient().CdiV1beta1().DataSources(namespace).Get(context.Background(), name, metav1.GetOptions{})
		if err != nil && !k8serrors.IsNotFound(err) {
			return err
		}

		if ds != nil {
			err = updateExistingDataSource(virtClient, ds)
		} else {
			err = createNewDataSource(virtClient, name, namespace)
		}
		if err != nil {
			return err
		}
	}

```

Reply:
Move calls to `setDefaultInstancetypeLabels()` into `updateExistingDataSource` and `createNewDataSource`.

Reply:
This also works.